astronomer/airflow/version_check/update_checks.py

```diff
@@ -24,19 +25,76 @@
 from airflow.utils.db import create_session
 from airflow.utils.log.logging_mixin import LoggingMixin
 from airflow.utils.timezone import utcnow
-from airflow.www import auth
+from functools import wraps
 
 try:
     from airflow.www_rbac.decorators import action_logging
 except ImportError:
     from airflow.www.decorators import action_logging
 
 
+T = TypeVar("T", bound=Callable)
+
 # Code is placed in this file as the default Airflow logging config shows the
 # file name (not the logger name) so this prefixes our log messages with
 # "update_checks.py"
 
 
+def has_access_(permissions: Sequence[tuple[str, str]]) -> Callable[[T], T]:
+    method: str = permissions[0][0]
+    resource_type: str = permissions[0][1]
+
+    from airflow.utils.net import get_hostname
+    from airflow.www.extensions.init_auth_manager import get_auth_manager
+
+    def decorated(*, is_authorized: bool, func: Callable, args, kwargs):
+        """
+        Define the behavior whether the user is authorized to access the resource.
+        :param is_authorized: whether the user is authorized to access the resource
+        :param func: the function to call if the user is authorized
+        :param args: the arguments of ``func``
+        :param kwargs: the keyword arguments ``func``
+        :meta private:
+        """
+        if is_authorized:
+            return func(*args, **kwargs)
+        elif get_auth_manager().is_logged_in() and not g.user.perms:
+            return (
+                render_template(
+                    "airflow/no_roles_permissions.html",
+                    hostname=get_hostname() if conf.getboolean("webserver", "EXPOSE_HOSTNAME") else "redact",
+                    logout_url=get_auth_manager().get_url_logout(),
+                ),
+                403,
+            )
+        else:
+            access_denied = conf.get("webserver", "access_denied_message")
+            flash(access_denied, "danger")
+        return redirect(get_auth_manager().get_url_login(next=request.url))
+
+    def has_access_decorator(func: T):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            return decorated(
+                is_authorized=get_auth_manager().is_authorized(method=method, resource_type=resource_type),
```

Comment:
Should this be `appbuilder.sm.check_authorization`?

Reply:
@ephraimbuddy I'm not entirely sure TBH. My understanding of this is limited. 
However `get_auth_manager().is_authorized()` depends on auth_manager and the other changes `appbuilder.sm.check_authorization` depend on the flask security_manager. I think auth_manger is a higher-level abstraction, so I slightly lean towards keeping existing changes. But no preference.